src/Uno.UI/UI/Xaml/UIElement.skia.cs

```diff
@@ -73,14 +73,14 @@ private void UpdateOpacity()
 			Visual.Opacity = Visibility == Visibility.Visible ? (float)Opacity : 0;
 		}
 
-		internal ShapeVisual Visual
+		internal BorderVisual Visual
```

Comment:
This is very much surprising. I don't think we should use `BorderVisual` for everything. Instead, I'd make it virtual and override where necessary, or, alternatively, do `if (this is IBorderInfoProvider)